### PR TITLE
APP-2999: Signing using env variables when provided

### PR DESCRIPTION
**Pull request by symphony-elias**
### Ticket
[APP-2999](https://perzoinc.atlassian.net/browse/APP-2999)

### Description
Use env variables to sign artifacts so that it can be used in the jenkins build

### Checklist
- [ ] Referenced a ticket in the PR title and in the corresponding section
- [ ] Filled properly the description and dependencies, if any
- [ ] Unit tests updated or added
- [ ] Javadoc added or updated
- [ ] Updated the documentation in [docs folder](../docs)

### CODE REVIEW

File: jenkins/Jenkinsfile-publish [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    def branch = env.PARAM_BUILD_BRANCH ?: 'master'
+
+    withCredentials([
+        file(credentialsId: 'JAVA_SDK_GPG_KEY', variable: 'ORG_GRADLE_PROJECT_KEY'),
+        string(credentialsId: 'JAVA_SDK_GPG_PASSPHRASE', variable: 'ORG_GRADLE_PROJECT_PASSPHRASE')
+    ]) {
+    
+      withEnv([
+        "PROJECT_TYPE=java",
+        "USE_OPENJDK11=false",
```
Comment (symphony-youri) [inline jenkins/Jenkinsfile-publish:23]: given we are not using much of pipeline lib we could get rid of JDK, SONAR variables

File: jenkins/Jenkinsfile-publish [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    
+        stage('Git Checkout') {
+            gitCheckout()
+        }
+        
+        stage('Build project') {
+            sh './gradlew clean build'
+        }
+        
+        stage('Check jar file') {
```
Comment (symphony-youri) [inline jenkins/Jenkinsfile-publish:38]: do we need to keep that or is it just for now to debug?

Comment (symphony-youri) [reply jenkins/Jenkinsfile-publish:38]: (and the target/ find won/t work, Gradle uses build/ for its output)

File: jenkins/Jenkinsfile-publish [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            sh 'ls -la ./symphony-bdk-legacy/symphony-api-client-java/target'
+            sh 'find . -type f -regex ".*/target/symphony-api-client-java-.*.jar" -exec echo {} \\;'
+        }
+        
+        stage('Deploy') {
+            sh './gradlew -PsymphonyRepoUrl=$PARAM_SYMPHONY_REPO -PsymphonyRepoUser=dev-services -PsymphonyRepoPassword=$TBD publish'
```
Comment (symphony-youri) [inline jenkins/Jenkinsfile-publish:44]: TBD -> do you know how we can retrieve it?

Comment (symphony-elias) [reply jenkins/Jenkinsfile-publish]: After checking, I saw we had credentials in jenkins

File: jenkins/Jenkinsfile-publish (hunk 2 above)
Comment (symphony-elias) [reply jenkins/Jenkinsfile-publish]: Renoved

**Pull request closed without merging**